blockchain/blockdao/blockdao.go

```diff
@@ -620,25 +640,20 @@ func (dao *blockDAO) putBlock(blk *block.Block) error {
 }
 
 // deleteTipBlock deletes the tip block
-func (dao *blockDAO) deleteTipBlock() (*block.Block, error) {
+func (dao *blockDAO) deleteTipBlock() error {
```

Comment:
this is good catch, since we move indexer back into DAO, no need to return the deleted block